packages/jaeger-ui/src/components/TracePage/CriticalPath/utils/findLastFinishingChildSpanId.tsx

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2023 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Span, Trace } from '../../../../types/trace';
+
+/**
+ * This function finds the Last Finishing Child (LFC) span of a given current span
+ *  Example:
+ * |-------------spanA--------------|
+ *    |--spanB--|    |--spanC--|
+ * The LFC of spanA would be spanC, as it finishes last among its child spans.
+ *
+ * However, if a child span (like spanC) doesn't have any child spans of its own,
```

Comment:
* I don't see anything in the function's code that would inspect `C.children`
* why recursive children are even relevant to the function as stated in its name?
* what is the role of spawnTime parameter?
* I see no sorting logic in the code, so how does this actually ensure that this is the LAST child?

Reply:
2.)The function's name might initially suggest that it should deal with recursive children. However, implementation, is to focus on finding the last finishing child span among the immediate children of the current Span. Recursive children are not directly relevant in this function, as it's looking only one level deep into the child spans.

4.) There is no need to sort because already we sorted childSpanIds based on endTime in transformData only so we can ensure that actually it is LAST child
https://github.com/jaegertracing/jaeger-ui/blob/eb28e9ebd074d1640d61e41e1e32cacecf462120/packages/jaeger-ui/src/model/transform-trace-data.tsx#L138-L147

Reply:
`Role of spawnTime:` I explained in this comment ,Please take a look and let me know if you need any further clarifications.
https://github.com/jaegertracing/jaeger-ui/pull/1532#discussion_r1285084429